src/metamorpho/MetamorphoSnippets.sol

```diff
@@ -15,58 +15,80 @@ import {ERC20} from "@openzeppelin/token/ERC20/ERC20.sol";
     }
 
-    function totalSharesUserVault(address user) public view returns (uint256 totalSharesUser) {
-        totalSharesUser = vault.balanceOf(user);
+    /// @notice Returns the total shares balance of a `user` on a metamorpho `vault`.
+    /// @param vault The address of the metamorpho vault.
+    /// @param user The address of the user.
+    function totalSharesUserVault(address vault, address user) public view returns (uint256 totalSharesUser) {
+        totalSharesUser = IMetaMorpho(vault).balanceOf(user);
     }
 
-    // The following function will return the current supply queue of the vault
-    function supplyQueueVault() public view returns (Id[] memory supplyQueueList) {
-        uint256 queueLength = vault.supplyQueueLength();
+    /// @notice Returns the supply queue a metamorpho `vault`.
+    /// @param vault The address of the metamorpho vault.
+    function supplyQueueVault(address vault) public view returns (Id[] memory supplyQueueList) {
+        uint256 queueLength = IMetaMorpho(vault).supplyQueueLength();
         supplyQueueList = new Id[](queueLength);
+
         for (uint256 i; i < queueLength; ++i) {
-            supplyQueueList[i] = vault.supplyQueue(i);
+            supplyQueueList[i] = IMetaMorpho(vault).supplyQueue(i);
         }
+
         return supplyQueueList;
     }
 
-    // // The following function will return the current withdraw queue of the vault
-    function withdrawQueueVault() public view returns (Id[] memory withdrawQueueList) {
-        uint256 queueLength = vault.supplyQueueLength();
+    /// @notice Returns the withdraw queue a metamorpho `vault`.
+    /// @param vault The address of the metamorpho vault.
+    function withdrawQueueVault(address vault) public view returns (Id[] memory withdrawQueueList) {
+        uint256 queueLength = IMetaMorpho(vault).supplyQueueLength();
         withdrawQueueList = new Id[](queueLength);
+
         for (uint256 i; i < queueLength; ++i) {
-            withdrawQueueList[i] = vault.withdrawQueue(i);
+            withdrawQueueList[i] = IMetaMorpho(vault).withdrawQueue(i);
         }
+
         return withdrawQueueList;
     }
 
-    function capMarket(MarketParams memory marketParams) public view returns (uint192 cap) {
+    /// @notice Returns the supply cap of a market on a metamorpho `vault`.
+    /// @param vault The address of the metamorpho vault.
+    /// @param marketParams The morpho blue market.
+    function capMarket(address vault, MarketParams memory marketParams) public view returns (uint192 cap) {
         Id id = marketParams.id();
-        cap = vault.config(id).cap;
+        cap = IMetaMorpho(vault).config(id).cap;
     }
 
+    /// @notice Returns the current APR (Annual Percentage Rate) of a morpho blue market.
+    /// @param marketParams The morpho blue market parameters.
+    /// @param market The morpho blue market state.
     function supplyAPRMarket(MarketParams memory marketParams, Market memory market)
```

Comment:
I'm discussing whether this APR is what we want to promote: https://morpholabs.slack.com/archives/C02N7CZ088N/p1701249695488329?thread_ts=1701249519.364069&cid=C02N7CZ088N

Reply:
It seems that the APR is of no interest, because it's never accounted raw like this in Blue

We could instead promote the APY:
- borrow: `borrowRate.wTaylorCompounded(1)`
- supply: `borrowRate.wTaylorCompounded(1).wMulDown(utilization).wMulDown(WAD - market.fee)`